tests/apple-pay/applepay.spec.ts

```diff
@@ -0,0 +1,33 @@
+import { test, PaymentStepAsGuestPreset } from '../';
+import { applepay, internalOrder, order, orderPayment, payments, validateMerchantResponse } from './ApplePayTestMockResponse';
+
+test.describe('ApplePay', () => {
+    test('ApplePay in payment step', async ({ assertions, checkout, page }) => {
+        // Testing environment setup
+        await page.addInitScript({ path: './tests/apple-pay/ApplePaySessionMockObject.js' });
+
+        const storeUrl = 'https://my-dev-store-117450812.store.bcdev/';
+        const responseProps = { status: 200, contentType: 'application/json' };
+        await checkout.use(new PaymentStepAsGuestPreset(storeUrl));
+        await checkout.create('ApplePay in Payment Step', storeUrl);
+        await checkout.route(/order-confirmation.*/, './tests/support/orderConfirmation.ejs', { orderId: '124' });
+        await page.route(/.*\/api\/storefront\/payments\?cartId=.*/, route => route.fulfill({ ...responseProps, body: payments }));
```

Comment:
Actually, I think all these endpoints should be automatically mocked via PollyJS. Is there a reason why it needs to mocked manually?

Reply:
@bc-peng

Reply:
`playwright` is unable to use `ApplePay` to place a legitimate order because `ApplePay` runs outside of the area where `playwright` can click. 

In order to fill in the responses that should be produced by interactions with `ApplePay`, for example, `/api/storefront/orders/124`, we mocked some API responses.

`/api/storefront/payments\payments` can be removed,  I mocked it to only return `ApplePay` in order to speed up the payment section loading.

Reply:
I think we can unless I'm mistaken. Although in our test we're using the fake `ApplePaySession` instead of the real `ApplePaySession`, we are still making HTTP calls to our APIs therefore they can be recorded by PollyJS and replayed in subsequent runs.

Reply:
@davidchin I understand your concern that if the mocked `ApplePaySession` can actually make a valid API call, we shouldn't mock the call.

In this case, I think that 3 out of 7 mockups can be omitted:

- Loading payment methods.
- Loading `ApplePay`.
- Validating a merchant with `ApplePay` server.

Safari's bulit-in `ApplePay` JS generates a token during authorizing payment. We probably won't be able to reproduce it on Chrome. 

As a result, we have to mock everything from authorizing payment to the end.

https://developer.apple.com/documentation/passkit/apple_pay/payment_token_format_reference
<img width="946" alt="Screen Shot 2022-08-05 at 3 52 22 pm" src="https://user-images.githubusercontent.com/88361607/183010575-6a1b9d31-09eb-4044-b313-cce23e2f5c40.png">

Reply:
I've removed the mock routes that Peng has suggested.